internal/ingress/metric/main.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metric
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// Namespace default metric namespace
+var Namespace = "nginx_ingress_controller"
+
+// Registry default metric registry
+var Registry = prometheus.DefaultGatherer
```

Comment:
I highly recommend not using a global style metrics registry, we shouldn't have done this in the first place in Prometheus but we can't undo that now :slightly_smiling_face: . I recommend injecting a registry wherever you need it. For a start you can even pass the global prometheus registry but over time you can make sure to always pass a registry and eventually remove the global registry all together. The problem with the global metrics registry for example is that arbitrary libraries can register metrics via init functions and in the worst case even register metrics that you are intending to register which then panics. Or the minikube example where all kubernetes components are run as one process which causes the apiserver to expose scheduler metrics, which is odd at best :slightly_smiling_face: .

Reply:
@brancz thank you for this info. I will replace this with a new registry before merging